src/test/java/org/junitpioneer/jupiter/CartesianProductTestSetsTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.util.Lists.list;
+
+import java.util.List;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("The sets for CartesianProductTest")
+class CartesianProductTestSetsTest {
```

Comment:
Personally, I'd put these in a `@Nested` class in `CartesianProductTestExtensionTests`.

Reply:
Also, this test never runs because we only run test classes that match the pattern: `.*Tests`.

Reply:
Ah, thank you! Never realized this. This is also the reason why Sonar isn't happy.

Regarding `@Nested`: `CartesianProductTestExtensionTests` is already quite big, so I thought having a separate class helps. But I can change that.

Reply:
@Michael1993 converted the test to a nested one. We "now" have some redundant test cases since e.g. `RedundantInputSetTestCases` does essentially the same, just via a different test interface.

Reply:
> Regarding `@Nested`: `CartesianProductTestExtensionTests` is already quite big, so I thought having a separate class helps. But I can change that.

I'm totally fine with splitting large test classes into multiple files and don't overuse `@Nested`.

Reply:
> @Michael1993 converted the test to a nested one. We "now" have some redundant test cases since e.g. `RedundantInputSetTestCases` does essentially the same, just via a different test interface.

When they use different interfaces they go different ways, don't they?